capstone-frontend/src/components/ListPage.jsx

```diff
@@ -0,0 +1,90 @@
+import React, { Component } from 'react';
+
+class ListPage extends Component {
+
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      gBooks: [],
+      loading: true,
+      empty: false
+    }
+  }
+
+
+  fetchBooks = async () => {
+
+    const userID = window.localStorage.getItem("userID");
+
+    console.log("Fetch Books LIST PAGE")
+
+    const bookLists = await fetch(`/api/booklist?userID=${userID}`, {
+      method: "GET",
+    }).then(resp => resp.json());
+
+    console.log("Booklist", bookLists)
+
+    const bookList = Array.from(bookLists).filter(bookList => (bookList.id === this.props.match.params.id))[0].gbookIDs;
+
+    console.log("Booklist", bookList)
+
+    if (bookList.length === 0) {
+      this.setState({loading: false, empty: true});
+      return;
+    }
+
+    const gBooks = [];
+
+    await Promise.all(bookList.map(async (gBookID) => {
+      const gBook = await fetch(`https://www.googleapis.com/books/v1/volumes/${gBookID}`).then(response => response.json())
+      gBooks.push(gBook);
+    }))
+
+    this.setState({ gBooks, loading: false });
+    console.log(gBooks)
+  }
+
+  componentDidMount() {
+    this.fetchBooks();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.match.params.id !== prevProps.match.params.id) {
+      this.setState({ loading: true })
+      this.fetchBooks();
+    }
+  }
+
+  convertToHttps = (imgURL) => {
+    return imgURL.slice(0, 4) + "s" + imgURL.slice(4);
+  }
+
+  render() {
+    return this.state.loading ? (<h1 className="text-center mt-4">Loading...</h1>) : (this.state.empty ? (<h1 className="text-center mt-4">Booklist has No Books</h1>) : (
+      <div className="text-center mt-4">
+        {
+          this.state.gBooks.map(gBook =>
+            <div key={gBook.id + Math.random().toString(36).substr(2, 9)}>
```

Comment:
Without the addition of the `Math.random()...` segment does this fail if multiple of the same gBook are in a list or if the gBook was present in another list you viewed?

I think we should discuss a way to not add randomness for this key calculation

Reply:
+1, why don't we use the gbookID as the key? Those should already be unique, and a more obvious choice since the books come with them.

Reply:
@antoniolinhart came up with a good solution.